Bio/SeqIO/Interfaces.py

```diff
@@ -141,25 +147,91 @@ def clean(self, text):
+    def write_header(self):
+        """Write the file header to the output file."""
+        pass
+        ####################################################
+        # You SHOULD implement this method in the subclass #
+        # if the file format defined a file header.        #
+        ####################################################
+
+    def write_footer(self):
+        """Write the file footer to the output file."""
+        pass
+        ####################################################
+        # You SHOULD implement this method in the subclass #
+        # if the file format defined a file footer.        #
+        ####################################################
+
+    def write_record(self, record):
+        """Write a single record to the output file.
+
+        record - a SeqRecord object
+        """
+        raise NotImplementedError("This method should be implemented")
+        ####################################################
+        # You SHOULD implement this method in the subclass #
+        # for sequential file formats.                     #
+        ####################################################
+
+    def write_records(self, records, maxcount=None):
+        """Write records to the output file, and return the number of records.
 
         records - A list or iterator returning SeqRecord objects
+        maxcount - The maximum number of records allowed by the
+        file format, or None if there is no maximum.
+        """
+        count = 0
+        if maxcount is None:
+            for record in records:
+                self.write_record(record)
+                count += 1
+        else:
+            for record in records:
+                if count == maxcount:
+                    if maxcount == 1:
+                        raise ValueError("More than one sequence found")
+                    else:
+                        raise ValueError(
+                            "Number of sequences is larger than %d" % maxcount
+                        )
+                self.write_record(record)
+                count += 1
+        return count
 
-        Should return the number of records (as an integer).
+    def write_file(self, records, mincount=0, maxcount=None):
+        """Write a complete file with the records, and return the number of records.
 
-        This method can only be called once.
+        records - A list or iterator returning SeqRecord objects
         """
-        # Note when implementing this, your writer class should NOT close the
-        # file at the end, but the calling code should.
-        raise NotImplementedError("This object should be subclassed")
-        #####################################################
-        # You SHOULD subclass this                          #
-        #####################################################
+        ####################################################
+        # You SHOULD implement this method in the subclass #
```

Comment:
Remove this comment, or at least rephrase. When would you want to implement this?